File: frontend/src/app/features/alerts/models/alert-query-parameters.ts
import { AlertSeverity } from "./alert-severity-enum";
import { QueryParameters } from "../../../core/services/rest-api/paging/query-parameters";

export interface AlertQueryParameters extends QueryParameters {
  withAppId?: string;
  ofSeverity?: AlertSeverity;
  occurredBefore?: string;
  occurredAfter?: string;
}

File: http-client/MCS.WatchTower.WebApi.DataTransferObjects/QueryParameters/AlertQueryParameters.cs
#nullable enable
using MCS.WatchTower.WebApi.DataTransferObjects.Enumerations;

namespace MCS.WatchTower.WebApi.DataTransferObjects.QueryParameters;

public record AlertQueryParameters(
    string? WithAppId = null,
    AlertSeverity? OfSeverity = null,
    DateTime? OccurredBefore = null,
    DateTime? OccurredAfter = null,
    string? SearchTerm = null,
    string? OrderBy = null,
    int? Page = null,
    int? PageSize = null
) : Paging.QueryParameters(SearchTerm, OrderBy, Page, PageSize);
